builtin/logical/pki/crl_util.go

```diff
@@ -399,6 +563,36 @@ func revokeCert(ctx context.Context, b *backend, req *logical.Request, serial st
 				return nil, fmt.Errorf("error encountered during CRL building: %w", crlErr)
 			}
 		}
+	} else {
+		// Regardless of whether or not we've presently enabled Delta CRLs,
+		// we should always write the Delta WAL in case it is enabled in the
+		// future. We could trigger another full CRL rebuild instead (to avoid
+		// inconsistent state between the CRL and missing Delta WAL entries),
+		// but writing extra (unused?) WAL entries versus an expensive full
+		// CRL rebuild is probably a net wash.
+		//
+		// Currently we don't store any data in the WAL entry.
+		var walInfo deltaWALInfo
+		walEntry, err := logical.StorageEntryJSON(deltaWALPath+normalizeSerial(serial), walInfo)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create delta CRL WAL entry")
+		}
+
+		if err = req.Storage.Put(ctx, walEntry); err != nil {
+			return nil, fmt.Errorf("error saving delta CRL WAL entry")
+		}
+
+		// In order for periodic delta rebuild to be mildly efficient, we
+		// should write the last revoked delta WAL entry so we know if we
+		// have new revocations that we should rebuild the delta WAL for.
+		lastRevSerial := lastWALInfo{Serial: serial}
+		lastWALEntry, err := logical.StorageEntryJSON(deltaWALLastRevokedSerial, lastRevSerial)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create last delta CRL WAL entry")
+		}
+		if err = req.Storage.Put(ctx, lastWALEntry); err != nil {
+			return nil, fmt.Errorf("error saving last delta CRL WAL entry")
+		}
```

Comment:
Should this not be protected by a alreadyRevoked == false?

Reply:
Yeah, that's a good point. We don't really want to write it to the delta again.